sys/syz-extract/netbsd.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2017 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/google/syzkaller/pkg/compiler"
+)
+
+type netbsd struct{}
+
+func (*netbsd) prepare(sourcedir string, build bool, arches []string) error {
+	if sourcedir == "" {
+		return fmt.Errorf("provide path to kernel checkout via -sourcedir flag (or make extract SOURCEDIR)")
+	}
+	if !build {
+		return fmt.Errorf("netbsd requires -build flag")
+	}
+	return nil
+}
+
+func (*netbsd) prepareArch(arch *Arch) error {
+	if err := os.Symlink(filepath.Join(arch.sourceDir, "sys", "arch", "amd64", "include"),
+		filepath.Join(arch.buildDir, "machine")); err != nil {
+		return fmt.Errorf("failed to create link: %v", err)
+	}
+	if err := os.Symlink(filepath.Join(arch.sourceDir, "sys", "arch", "x86", "include"),
+		filepath.Join(arch.buildDir, "x86")); err != nil {
+		return fmt.Errorf("failed to create link: %v", err)
+	}
+	return nil
+}
+
+func (*netbsd) processFile(arch *Arch, info *compiler.ConstInfo) (map[string]uint64, map[string]bool, error) {
+	args := []string{
+		"-fmessage-length=0",
+		"-nostdinc",
+		"-D_KERNEL",
+		"-D__BSD_VISIBLE=1",
```

Comment:
There is no such thing in NetBSD headers.

Reply:
I just checked and verified that this is a FreeBSD thing. Maybe you intend `_NETBSD_SOURCE`, which is explicit unless someone defines `_POSIX_SOURCE` or similar.

Reply:
Actually, there were a lot of things that I had to verify/change and I missed a few of them. Thanks for pointing out.

Reply:
> I would suggest to remove (comment out) all of these defines (including __BSD_VISIBLE/_NETBSD_SOURCE/etc) and then try to execute extract and add macros back as needed.

Thanks! I’ll do that and commit the necessary changes. 😄